### PR TITLE
Refresh token on expiration

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1894)
Our auth token is expiring during the process of pulling down all DRS objects. 

## This PR
* Checks if the token has expired, and refreshes if needed.

## Checklist
- [x] Documentation has been updated as needed.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_orchestration/resources/config/target_hca_dataset.py [hunk 1]
```diff
@@ -32,19 +32,20 @@ def build_new_target_hca_dataset(init_context: InitResourceContext) -> Optional[
 
     creation_date = datetime.now().strftime("%Y%m%d")
     env = init_context.resource_config['env']
-    target_hca_dataset_name = f"hca_{env}_{hca_project_id.hex.replace('-', '')}__{creation_date}"
+    target_hca_dataset_prefix = f"hca_{env}_{hca_project_id.hex.replace('-', '')}"
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/resources/config/target_hca_dataset.py:35]: We were always going to create a new dataset if the day rolls over (i.e, a copy on 8/30 would create a dataset, and a subsequent re-copy on 9/1 would create an all-new dataset). This checks for an existing dataset with the needed project_id and re-uses it if present.

File: orchestration/dagster_orchestration/hca_orchestration/resources/config/target_hca_dataset.py [hunk 2]
```diff
@@ -32,19 +32,20 @@ def build_new_target_hca_dataset(init_context: InitResourceContext) -> Optional[
...
 
     if not result:
-        init_context.log.info(f"Target dataset {target_hca_dataset_name} not found, creating")
+        init_context.log.info(f"Target dataset with prefix {target_hca_dataset_prefix} not found, creating")
         return data_repo_service.create_dataset(
             target_hca_dataset_name,
             env,
             data_repo_profile_ids[env],
-            {"aherbst@broadinstitute.org"},
+            {"monster-dev@dev.test.firecloud.org"},
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/resources/config/target_hca_dataset.py:48]: 🤦

File: orchestration/dagster_orchestration/hca_orchestration/solids/copy_project/subgraph_hydration.py [hunk 3]
```diff
@@ -126,8 +126,14 @@ def hydrate_subgraphs(context: AbstractComputeExecutionContext) -> set[DataFileE
         s.headers["Authorization"] = f"Bearer {creds.token}"
 
         for cnt, (drs_object, drs_host) in enumerate(drs_objects.items(), start=1):
+            if creds.expired:
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/solids/copy_project/subgraph_hydration.py:129]: This is the actual fix.

**Review by quazi-broad — APPROVED** [review]
changes look good to me, hopefully this resolves a few of the failed dev refresh runs

**Pull request merged**